# Generated by Django 4.2.22 on 2025-07-26 00:16

from django.db import migrations, models


class Migration(migrations.Migration):
    dependencies = [
        ("posthog", "0806_scheduledchange_failure_count_and_more"),
    ]

    operations = [
        migrations.CreateModel(
            name="PersonPropertiesSizeViolation",
            fields=[
                ("id", models.BigAutoField(primary_key=True, serialize=False)),
                (
                    "person_id",
                    models.BigIntegerField(db_index=True, help_text="ID of the person with the size violation"),
                ),
                ("properties_size_bytes", models.BigIntegerField(help_text="Size of the properties field in bytes")),
                (
                    "properties_size_human",
                    models.CharField(help_text="Human-readable size representation", max_length=20),
                ),
                ("detected_at", models.DateTimeField(auto_now_add=True, db_index=True)),
                (
                    "operation_type",
                    models.CharField(
                        choices=[("INSERT", "Insert"), ("UPDATE", "Update")],
                        help_text="Database operation that triggered the violation",
                        max_length=10,
                    ),
                ),
            ],
            options={
                "db_table": "posthog_person_properties_size_violation",
                "indexes": [
                    models.Index(fields=["person_id", "detected_at"], name="posthog_per_person__54cb86_idx"),
                    models.Index(fields=["properties_size_bytes"], name="posthog_per_propert_607ca5_idx"),
                ],
            },
        ),
    ]
